### PR TITLE
Enable lead-time-aware regression without CE loss

### PR DESCRIPTION
**Pull request by tge25**
# Enable lead-time-aware regression without CE loss

## Description
Adds support for the feature requested in #918 

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.

### CODE REVIEW

File: physicsnemo/metrics/diffusion/loss.py [hunk 1]
```diff
@@ -379,6 +379,7 @@ def __call__(
         net: torch.nn.Module,
         img_clean: torch.Tensor,
         img_lr: torch.Tensor,
+        lead_time_label: Optional[torch.Tensor] = None,
```
Comment (CharlelieLrt) [inline physicsnemo/metrics/diffusion/loss.py:382]: Could you add a unit test for the `ResidualLoss` with a `lead_time_label` provided? It's probably sufficient to just copy paste one of the existing tests for `ResidualLoss` and add an argument.

**Review by CharlelieLrt — CHANGES_REQUESTED** [review]
LGTM, we just need a unit test for the additional argument passed to the loss.

File: physicsnemo/metrics/diffusion/loss.py (hunk 1 above)
Comment (tge25) [reply physicsnemo/metrics/diffusion/loss.py]: unit test for the ResidualLoss with a lead_time_label has been added. Thanks

**Review by CharlelieLrt — APPROVED** [review]
LGTM
